core/scripts/chaincli/handler/keeper.go

```diff
@@ -90,16 +92,36 @@ func (k *Keeper) DeployKeepers(ctx context.Context) {
 }
 
 // DeployRegistry deploys a new keeper registry.
-func (k *Keeper) DeployRegistry(ctx context.Context) {
+func (k *Keeper) DeployRegistry(ctx context.Context, verify bool) {
+	if verify {
+		if k.cfg.RegistryVersion != keeper.RegistryVersion_2_1 && k.cfg.RegistryVersion != keeper.RegistryVersion_2_0 {
+			log.Fatal("keeper registry verification is only supported for version 2.0 and 2.1")
+		}
+		if k.cfg.ExplorerAPIKey == "" || k.cfg.ExplorerAPIKey == "<explorer-api-key>" || k.cfg.NetworkName == "" || k.cfg.NetworkName == "<network-name>" {
+			log.Fatal("please set your explore API key and network name in the .env file to verify the registry contract")
+		}
+
+		// Get the current working directory
+		currentDir, err := os.Getwd()
```

Comment:
can we move this check to `changeToContractsDirectory()`?

Reply:
This check is optional, it serves the purpose of "fail fast". In `changeToContractsDirectory()` we have checks defined to make sure we are or we are able to get to the right directory(contracts). This check is run only if verify flag is on. I can remove this check from here if you feel that is better.

Reply:
ah I see now. Might be irrelevant now if we support calling from any directory?

Reply:
Not really. As you can see from [here](https://github.com/smartcontractkit/chainlink/pull/9421/files#r1218379162), being in the contracts directory is required to run hardhat verification, but moving to it is that smart.

Based on my understanding, most likely people run cli commands from core/script/chaincli directory because it requires the .env file in that folder. People can also run cli from anywhere if they provide their own config, but i guess it is not easy/necessary to have another config.

Given that we need to move to the contracts folder from current working directory, I need to have this kind of checks, here(optional for fail fast) and `changeToContractsDirectory()` to ensure we are able to get to the right place to run hardhat verify command.